shared/extension/behaviour/commands.hpp

```diff
@@ -35,36 +35,38 @@ namespace extension::behaviour::commands {
      * The classification type of the Provider
      */
     enum class ProviderClassification {
-        /// NORMAL Providers are the usual Providers that execute in a loop if needed
-        NORMAL,
-        /// ENTERING Providers are the first point of entry for a type before the main NORMAL reaction runs
-        ENTERING,
-        /// LEAVING Providers are executed as a module is relinquishing control
-        LEAVING,
+        /// Provide providers are typical providers that run when given a task
+        PROVIDE,
+        /// Leave providers are able to be forced to run when a higher priority task needs a causing it makes
+        LEAVE,
+        /// Start providers run once when a provider group first gets a task
+        START,
+        /// Stop providers run once there are no tasks running for any provider of this group
+        STOP
```

Comment:
Are Leaving and Stop similar, but Stop is like Start in that it runs once and doesn't Cause anything, and then Leaving will keep running until it's happy and its Cause is satisfied?

Reply:
Leave is a bad name for what it has turned into. If you have a better name for it I'll happily change it.

As an example of why Leaving is a thing

You have a walk command with two normal `Provide` commands one is a normal walk and the other has a `Causing` for `Standing` that stops the walk and stands still.

You have a second module for `Look` that only works when the robot is in the `Standing` state that has a lower challenge priority than walk. It can still force the walk to start using the `Provide` statement that would stop the walk and make it stand. Which is not ideal.

Leaving providers are only able to be bullied into that state by a module with higher challenge priority. So if the look command were a higher priority it would be able to force it into a `Leave` provider. But if it was lower priority it would only be able to force it into other `Provide` statements.

So what you should do is to make the walk provider that makes the walk stop a `Leave` provider. That way it can only be forced into the stop state by another module that wants something the walk engine can do that would go against the main task (walking). So in a way it's "Leaving" although it might never actually stop running walk commands. Look might finish and the walk will go back to normal without ever "Leaving"

 and the other just walks for example a `Walk` has a `Provide` that causes `Standing` and one that just walks then another module with lower priority that is running (say a look module wants to be standing still) can only run when